gammapy/cube/new.py

```diff
@@ -287,13 +287,97 @@ def process_obs(self, obs):
         )
 
         background_obs_map = make_map_background_fov(
-            acceptance_obs_map, count_obs_map, exclusion_mask_cutout,
+            acceptance_obs_map, counts_obs_map, exclusion_mask_cutout,
         )
 
-        self._add_cutouts(cutout_slices, count_obs_map, expo_obs_map, background_obs_map)
+        return cutout_slices, counts_obs_map, expo_obs_map, background_obs_map
 
-    def _add_cutouts(self, cutout_slices, count_obs_map, expo_obs_map, acceptance_obs_map):
+    def _add_cutouts(self, cutout_slices, counts_obs_map, expo_obs_map, acceptance_obs_map):
         """Add current cutout to global maps."""
-        self.count_map.data[cutout_slices] += count_obs_map.data
+        self.counts_map.data[cutout_slices] += counts_obs_map.data
         self.exposure_map.data[cutout_slices] += expo_obs_map.quantity.to(self.exposure_map.unit).value
         self.background_map.data[cutout_slices] += acceptance_obs_map.data
+
+    def process_obs(self, obs):
+        """
+        This will process a single observation and add it to the global cutout image.
+        This function may be removed after discussions.
+
+        """
+        cs, count, exp, back = self.process_single_obs(obs)
+        self._add_cutouts(cs, count, exp, back)
+
+
+    def run(self, obs_list):
+        """
+        Run MapMaker for a list of observations to create
+        stacked counts, exposure and background maps
+
+        Parameters
+        --------------
+        obs_list: `~gammapy.data.ObservationList`
+            List of observations
+
+        Returns
+        -----------
+        maps: dict of stacked counts, background and exposure maps.
+        """
+
+        from astropy.utils.console import ProgressBar
+
+        for obs in ProgressBar(obs_list):
+            cs, count, exp, back = self.process_single_obs(obs)
+            self._add_cutouts(cs, count, exp, back)
+
+        maps = {
```

Comment:
Make this a property called e.g. "results" or "maps"?
Then callers could access it after processing some obs, or at the end.
You could still return it from `run` if you want.